File: SimulationUtils/hs1d/src_python/old/time_properties.py

# coding: utf-8

# In[ ]:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import random
from datetime import timedelta 


# In[ ]:

class time_properties()
   

    def __init__(self, tmin, tmax, Nt, unit):
        
        self.tmin = tmin
        self.tmax = tmax
        self.Nt = Nt
        self.t = self.tmin:(self.tmax-self.tmin)/(Nt-1):self.tmax
        self.unit = unit
        
        tmin = time_to_seconds(tmin, unit)
        tmax = time_to_seconds(tmax, unit)

    def get_tmin(self):
        tmin = self.tmin
    
    def get_tmax(self):
        tmax = self.tmax

    def get_unit(self):
        unit = self.unit

    def get_Nt(self):
        Nt=self.Nt
    
    def get_size(self):
        size_ = size(self.t)

    def [t, unit] = get_properties(self):
        t = self.t
        unit=self.unit

File: SimulationUtils/hs1d/src_matlab/process/tools/boussinesq_resolution_tools/time_properties.m
classdef time_properties
    properties(Access=private)
        tmin
        tmax
        Nt
        t
        unit
    end
    
    methods(Access=public)
        function obj=time_properties(tmin,tmax,Nt,unit)
            tmin=time_unit.time_to_seconds(tmin,unit);
            tmax=time_unit.time_to_seconds(tmax,unit);
            
            obj.tmin=tmin;
            obj.tmax=tmax;
            obj.Nt=Nt;
            obj.t=obj.tmin:(obj.tmax-obj.tmin)/(Nt-1):obj.tmax;
            obj.unit=unit;
        end
    
        function tmin=get_tmin(obj)
            tmin=obj.tmin;
        end
        
        function tmax=get_tmax(obj)
            tmax=obj.tmax;
        end
        
        function unit=get_unit(obj)
            unit=obj.unit;
        end
        
        function Nt=get_Nt(obj)
            Nt=obj.Nt;
        end
        
        function size_=get_size(obj)
            size_=size(obj.t);
        end
        
        function [t,unit]=get_properties(obj)
            t=obj.t;
            unit=obj.unit;
        end
    end
end